tests/shanghai/eip4895_withdrawals/test_withdrawals.py

```diff
@@ -721,3 +722,178 @@ def test_withdrawing_to_precompiles(
         ),
     ]
     blockchain_test(pre=pre, post=post, blocks=blocks)
+
+def test_store_withdrawal_values_in_contract(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    yul: YulCompiler,
+):
+    """Test that system transaction calldata is correctly formed."""
+    contract_address = pre.deploy_contract(
+        code=yul(
+            """
+            {
+                let size := calldatasize()
+                let words := div(add(size, 31), 32)
+                for { let i := 0 } lt(i, words) { i := add(i, 1) } {
+                    sstore(i, calldataload(mul(i, 32)))
+                }
+                stop()
+            }
+            """
+        )
```

Comment:
This is a bit tricker, but I would change it to `SSTORE` the calldatasize, and then calldataload a hard-coded amount of times:
```suggestion
        code=Op.SSTORE(0, Op.CALLDATASIZE) + sum(Op.SSTORE(i + 1, Op.CALLDATALOAD(i * 32)) for i in range(10))
```
And then modify the post accordingly.